### PR TITLE
Adds build for publishing to NPM

### PR DESCRIPTION
**Pull request by Xyfi**
## Summary

This PR can be summarized in the following changelog entry:
* Adds a build script for a release build.

## Relevant technical choices:

* Added babel for transpiling code.

## Test instructions

This PR can be tested by following these steps:

* Run `grunt publish`
* Inspect the 'dist' folde

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   ],
   "main": "index.js",
   "license": "GPL-3.0",
-  "version": "1.22.0",
+  "version": "1.21.0",
```
Comment (atimmer) [inline package.json]: I assume this is unintentional.

File: .travis.yml [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
 - yarn run check-coverage
-
+after_script:
+- grunt publish
 after_success:
 - yarn global add codeclimate-test-reporter
 - codeclimate-test-reporter < coverage/lcov.info
-
+before_deploy:
+- cd dist
```
Comment (IreneStr) [inline .travis.yml]: Please add comment why you're doing this

File: examples/browserified/index.html [hunk 3]
```diff
@@ -6,7 +6,7 @@
     <link rel='stylesheet' id='source-sans-pro-css'
           href='https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,600&amp;subset=latin'/>
     <link rel="stylesheet" href="./style.css">
-    <link rel="stylesheet" href="../../dist/yoast-seo.min.css">
+    <link rel="stylesheet" href="../shared/yoast-seo.min.css">
```
Comment (IreneStr) [inline examples/browserified/index.html]: Why this change?

File: .gitignore [hunk 4]
```diff
@@ -4,6 +4,8 @@ temp-spec/
 node_modules/
 out/
 coverage/
-dist/yoast-seo*
+dist/
+js/
```
Comment (IreneStr) [inline .gitignore]: Because we're now publishing on NPM, and we're going to direct people to using that instead of linking the github repo, the js folder can be removed. Please make sure `grunt build` on YoastSEO.js is not broken after removing the js folder.

File: package.json [hunk 5]
```diff
@@ -9,7 +9,7 @@
   ],
   "main": "index.js",
   "license": "GPL-3.0",
-  "version": "1.22.0",
+  "version": "1.22.4",
```
Comment (IreneStr) [inline package.json]: Is this the correct version to end up in develop?

File: examples/browserified/index.html (hunk 3 above)
Comment (Xyfi) [reply examples/browserified/index.html]: It was in the dist folder, which is a more logical to place the built files. Since this css files is only used in the examples I moved it to a shared folder in the examples folder.

File: package.json (hunk 5 above)
Comment (Xyfi) [reply package.json]: I accidentally published a few versions with bugs in them. You can't undo a publish, so that's why.

**Comment by IreneStr** [comment]
CR and acceptance :+1:

**Comment by Xyfi** [comment]
Removed `yarn run check-coverage` from `.travis.yml` to make the build succeed.
